tests/sla_tests.js

```diff
@@ -113,11 +116,27 @@ var getMockChannel = function() {
       var self = this;
       setTimeout(function() {
         if(validEndpoint === input.endpoint) {
-          self.emit('StasisStart', {channel: {id: self.id}}, {channel: self});
-          pastStasis = true;
+          self.emit('StasisStart', {channel: {id: self.id}}, self);
+          pastOriginate = true;
         }
       }, asyncDelay);
     };
+    this.hangup = function(cb) {
+      var self = this;
+      for (var i = 0; i < channels.length; i++) {
```

Comment:
You can use channels.filter() to filter out the hungup channel. It would make the code a lot more readable.